src/components/Contacts/ContactFilters/ContactFilters.mock.tsx

```diff
@@ -0,0 +1,59 @@
+import { MockedResponse } from '@apollo/client/testing';
+import {
+  ContactFiltersDocument,
+  ContactFiltersQuery,
+  ContactFiltersQueryVariables,
+} from '../../../../pages/accountLists/[accountListId]/ContactFilters.generated';
+import { gqlMock } from '../../../../__tests__/util/graphqlMocking';
+
+export const ContactFiltersQueryDefaultMocks = (
+  accountListId: string,
+): MockedResponse[] => {
+  const contactFiltersQuery = gqlMock<
+    ContactFiltersQuery,
+    ContactFiltersQueryVariables
+  >(ContactFiltersDocument, { variables: { accountListId } });
+
+  return [
+    {
+      request: {
+        query: ContactFiltersDocument,
+        variables: {
+          accountListId,
+        },
+      },
+      result: {
+        data: contactFiltersQuery,
+      },
+    },
+  ];
```

Comment:
Interesting combination of automocks and manual mocks :) I like it although I hope we can keep the normal usage a little less verbose.

`<GqlMockedProvider>` should work in place of `<MockedProvider>`. Currently you can't customize the mocked data (the bug I've been meaning to look at more). But I'm pretty sure customizing the data works with `gqlMock(...)`. This could be a good way to get around that issue or create more complex mocks that match variables or return different values for different variables or return specific errors.

If I can't figure out my bug, this could be a cool way to implement it ourselves :)

Reply:
Yeah my only reason for passing `accountListId` into the mocks is that the component itself requires you to pass `accountListId` into it, so I want the mock data to be consistent with the data being put into the component.  If the mocks were completely automated but we hardcode the `accountListId` that is passed into the component, there's a chance that they might not be equal to each other.  So I wanted the test or stories file to determine what the `accountlistId` should be, and pass it in here.

I can try replacing `MockedProvider` with `GqlMockedProvider` throughout this PR.

Reply:
If you want to assert that it was queried with certain values you can use:
https://github.com/SurveyMonkey/graphql-ergonomock#:~:text=Finally,%20you%20can%20spy%20on%20executed%20operations%20via%20the%20onCall%20prop.